assets/js/phoenix_live_view/live_socket.js

```diff
@@ -407,7 +407,7 @@ export default class LiveSocket {
           DOM.findPhxSticky(document).forEach(el => newMainEl.appendChild(el))
           this.outgoingMainEl.replaceWith(newMainEl)
           this.outgoingMainEl = null
-          callback && requestAnimationFrame(callback)
+          callback && requestAnimationFrame(() => callback(linkRef))
```

Comment:
alternate approach...would it be possible to not call commitPendingLink on line 405 until later in the callback? Possible problems with that would be all the lines above (406-409) get run even if another navigation event is currently taking place, but not sure of a good example of what that would look like.